src/main/kotlin/com/cognifide/gradle/aem/pkg/deploy/InstallResponse.kt

```diff
@@ -1,28 +1,43 @@
 package com.cognifide.gradle.aem.pkg.deploy
 
-class InstallResponse private constructor(private val rawHtml: String) : HtmlResponse(rawHtml) {
+import java.io.InputStream
+import java.util.regex.Pattern
+
+class InstallResponse constructor(private val rawHtml: String) : HtmlResponse(rawHtml) {
 
     override fun getErrorPatterns(): List<ErrorPattern> {
         return mutableListOf(
-                ErrorPattern(InstallResponseBuilder.PROCESSING_ERROR_PATTERN, true),
-                ErrorPattern(InstallResponseBuilder.ERROR_PATTERN, false)
+                ErrorPattern(PROCESSING_ERROR_PATTERN, true),
+                ErrorPattern(ERROR_PATTERN, false)
         )
     }
 
     override val status: Status
         get() {
             return when {
-                rawHtml.contains(InstallResponseBuilder.INSTALL_SUCCESS) -> Status.SUCCESS
-                rawHtml.contains(InstallResponseBuilder.INSTALL_SUCCESS_WITH_ERRORS) -> Status.SUCCESS_WITH_ERRORS
+                rawHtml.contains(INSTALL_SUCCESS) -> Status.SUCCESS
+                rawHtml.contains(INSTALL_SUCCESS_WITH_ERRORS) -> Status.SUCCESS_WITH_ERRORS
                 else -> Status.FAIL
             }
         }
 
     companion object {
 
-        fun from(html: String): InstallResponse {
+        const val INSTALL_SUCCESS = "<span class=\"Package imported.\">"
+
+        const val INSTALL_SUCCESS_WITH_ERRORS = "<span class=\"Package imported (with errors"
+
+        val ERROR_PATTERN: Pattern =
+                Pattern.compile("<span class=\"E\"><b>E</b>&nbsp;(.+\\s??.+)</span>")
+
+        val PROCESSING_ERROR_PATTERN: Pattern =
+                Pattern.compile("<span class=\"error\">(.+)</span><br><code><pre>([\\s\\S]+)</pre>")
+
+        fun from(input: InputStream): InstallResponse {
             return try {
-                InstallResponse(html)
+                //TODO replace empty object with proper logic
```

Comment:
what? :)

Reply:
Just a personal comment, because ugly workaround was used to check some logic.